### ISSUE TITLE
Accumulate task inputs

### ISSUE THREAD

**Issue by anotherbugmaster** [PrefectHQ/prefect#2493]
## Use Case
*Please provide a use case to help us understand your request in context*

It would be great to be able to accumulate multiple task calls and execute them all at once. Examples are neural networks and database queries: doing one request at a time we don't use resources efficiently and get ourselves a bottleneck.

## Solution
*Please describe your ideal solution*

The simplest solution is to implement something like [this](https://docs.celeryproject.org/en/3.1/reference/celery.contrib.batches.html).

We need to add `flush_every` and `flush_interval` parameters to the `Task` class.

## Alternatives
*Please describe any alternatives you've considered, even if you've dismissed them*

I've read about Task Loops, but they don't quite work because they just run the task with the same arguments multiple times, whereas we need to run job once, but with multiple arguments.

**Comment by jcrist** [comment]
Since prefect stores state (and if using cloud/server also posts state to a server) for every task, there is always going to be overhead per task. If you're running fast tasks in prefect, you might consider wrapping these into a bulk api and having prefect schedule them at a higher level:

```python
def my_fast_task(x):
    ...

def batched_fast_task(args):
    return [my_fast_task(x) for x in args]
```

We could implement batching in prefect, but that gets complicated to schedule when dealing with retries, etc... IMO I'd like to see more use cases that require changes (and can't be done with user-level batching) before going down this route.

For mapped tasks there's the suggestion in #2459 that I think we would be willing to support, would this be sufficient for your needs?

**Comment by anotherbugmaster** [comment]
Hi, I've stumbled upon this issue again)

The problem is that tasks inputs are being streamed to the app. More specific, I have a web app that receives requests which I need to process in batches to improve performance. I need to fire up a task either when there are more than `N` tasks or when more than `X` seconds have passed since last run.

This feature was implemented in Celery v3, but developers removed it since then:
https://docs.celeryproject.org/en/3.1/reference/celery.contrib.batches.html#celery-contrib-batches
